### PR TITLE
SISRP-31634 - Dynamic Display of DB2 Archive Transcript Link

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-31634

### CODE REVIEW

**Comment by davidlimcheng** [comment]
Looks solid, @redconfetti.  The only thing is that the design at [SISRP-31500](https://jira.berkeley.edu/browse/SISRP-31500) includes a Font-Awesome "red exclamation" icon before the message, which I don't see in the HTML here.  Is this something that still needs to be added, or is the design outdated?

File: app/models/my_academics/academic_plan.rb [hunk 1]
```diff
@@ -18,6 +19,27 @@ def get_plan_semesters(semesters)
...
+      attributes_response = HubEdos::StudentAttributes.new(user_id: @uid).get
+      attributes = attributes_response.try(:[], :feed).try(:[], 'student').try(:[], 'studentAttributes')
+      if attributes
+        legacy_attribute = attributes.find {|attr| attr.try(:[], 'type').try(:[], 'code') == '+R11'}
+        campus_solutions_id = CalnetCrosswalk::ByUid.new(user_id: @uid).lookup_campus_solutions_id
+        if legacy_attribute && campus_solutions_id
+          return {
+            code: legacy_attribute.try(:[], 'reason').try(:[], 'code'),
+            message: legacy_attribute.try(:[], 'reason').try(:[], 'formalDescription'),
+            link: CampusSolutions::Link.new.get_url('UC_CX_ARCH_TSCRPT_ADVSTDNT', {EMPLID: campus_solutions_id}).try(:[], :link)
```
Comment (lyttam) [inline app/models/my_academics/academic_plan.rb]: @redconfetti why not use the `LinkFetcher` mixin here, and take advantage of the built-in error handling?

Comment (davidlimcheng) [inline app/models/my_academics/academic_plan.rb]: ooo, +1.

**Comment by redconfetti** [comment]
You're absolutely right @davidlimcheng. Great catch.

File: app/models/my_academics/academic_plan.rb (hunk 1 above)
Comment (redconfetti) [inline app/models/my_academics/academic_plan.rb]: Thanks @lyttam

**Pull request merged**